### PR TITLE
Add package nss-mdns

### PR DESCRIPTION
**Pull request by adobrzhansky**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Adds package nss-mdns to SPECS-EXTENDED

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- [New package request: Provide nss-mdns package support](https://github.com/microsoft/CBL-Mariner/issues/7406)

###### Test Methodology

- local build

Reffering to a docs section i might need some help with that part

> While it is possible to build packages with all sources inside the repo, our policy is generally to have the source compressed and placed on the source server. Uploading to the source server can only be accomplished by a CBL-Mariner developer. Please request help in your PR for uploading your sources to the source server. To check the source server see [https://cblmarinerstorage.blob.core.windows.net/sources/core/< source tar >].

### CODE REVIEW

**Comment by adobrzhansky** [comment]
@microsoft-github-policy-service agree

File: SPECS-EXTENDED/nss-mdns/nss-mdns.spec [hunk 1]
```diff
@@ -0,0 +1,228 @@
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+
+Name: nss-mdns
```
Comment (mfrw) [inline SPECS-EXTENDED/nss-mdns/nss-mdns.spec:4]: Test Build Triggered:
- [PR-7407](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=489086&view=results)

File: SPECS-EXTENDED/nss-mdns/nss-mdns.spec [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+Version: 0.15.1
+Release: 11%{?dist}
+Summary: glibc plugin for .local name resolution
+
+License: LGPLv2+
+URL: https://github.com/lathiat/nss-mdns
+Source0: https://github.com/lathiat/nss-mdns/releases/download/v%{version}/nss-mdns-%{version}.tar.gz
```
Comment (mfrw) [inline SPECS-EXTENDED/nss-mdns/nss-mdns.spec:11]: It seems that we missed to upload the source to the blobstore.
Reference: [link](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=489086&view=logs&j=7909073b-e471-53c1-2c44-627e2efc0564&t=fbeef549-39ab-510c-faa3-78579a516b53)

Comment (adobrzhansky) [reply SPECS-EXTENDED/nss-mdns/nss-mdns.spec:11]: Unfortunately i have no access to the CI build results but i did mention that at the bottom of PR's description e.g. 
> Referring to a docs section i might need some help with that part

So i would like to ask some help with uploading the source

Comment (mfrw) [reply SPECS-EXTENDED/nss-mdns/nss-mdns.spec:11]: Can you please share the sources offline ?

**Comment by adobrzhansky** [comment]
I have no access to rerun that CI check [Check Package CGManifests / Check Package CGManifests](https://github.com/microsoft/CBL-Mariner/actions/runs/7628182838/job/20779597514)  but it seems like some internal Github issue, could i please ask someone to trigger it again?

**Comment by mfrw** [comment]
Closing & Re-Opening PR to trigger stuck github checks

File: SPECS-EXTENDED/nss-mdns/nss-mdns.spec [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+%install
+rm -rf $RPM_BUILD_ROOT
+%make_install
+
+
+%post
+%{?ldconfig}
+
+%posttrans
+authselect enable-feature with-mdns4 &> /dev/null || :
```
Comment (mfrw) [inline SPECS-EXTENDED/nss-mdns/nss-mdns.spec:53]: I have triggered a test build:
[PR-7407](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=494569&view=results)

A note about the the prior test builds, they do `build` the package, but fail on `authselect` as it is a rutime dependency.

Have we verified the working of this package post a build ?

Comment (adobrzhansky) [reply SPECS-EXTENDED/nss-mdns/nss-mdns.spec:53]: Package was installed with no issues the only thing was to add `extended repo` from where `authselect` would come.
install
All the modules after installation exists in the `lib` dir [reference](https://github.com/avahi/nss-mdns?tab=readme-ov-file#libraries)
libs
Plugin activation and proof of working [reference](https://github.com/avahi/nss-mdns?tab=readme-ov-file#activation)
resolving

**Review by mfrw — APPROVED** [review]
LGTM :)
Thanks @adobrzhansky 
/cc @eric-desrochers for awareness.

**Pull request merged**